frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+if (false) {
+    // won't run
+} else if (true) {
+    // will run
+}
+
+// that was a brace-less else statement with an if as its single
+// statement. cool, huh? you can then chain off of that single
+// if statement with another else, an so on
+if (false) {
+    // won't run
+} else if (true) {
+    // will run
+} else {
+    // won't run (last condition was true)
+}
+
+// recall our previous overviews of logical operators and truthy values -
+// they both apply here. the condition just has to be a boolean!
+if (true && false) {
+    // won't run
+}
+
+if ("foo") {
+    // will run (nonempty strings are truthy)
+}
+```
+
+JavaScript gives you a shorthand for simple conditionals called the ternary expression. They're broken up into three parts: a condition, a value to yield if the condition is true, and a value to yield if the condition is false.
+```js
+// ternaries look like this:
+// condition ? leftHand : rightHand
+
+let ifTrue = "left hand side!";
+let ifFalse = "right hand side!";
+// the condition is true, so the ternary yields the value on the left side of
+// the colon - ifTrue, which is "left hand side!"
+true ? ifTrue : ifFalse;
+// now the condition is false, so the ternary yields the value on the right side
+// of the colon - ifFalse, which is "right hand side!"
+false ? ifTrue : ifFalse;
+
+// we can use this to compress basic conditionals down to a single line
+let condition = "some condition";
+let message;
+
+if (condition) {
+    message = "nice, condition is true";
+} else {
+    message = "drats, condition is false";
+}
+message === "nice, condition is true";
+
+// you could bring that down to something like this using a ternary
+message = condition ? "nice, condition is true" : "drats, condition is false";
+message === "nice, condition is true";
+```
+
+### Arrays
+Arrays let you bundle multiple values into one data structure.
+```js
+// array literals are delimited by square brackets ([])
+[0, 1, 2, 3]
+// elements don't have to be the same type
+let arr = ["value zero", "value one", 2, false];
+// elements are accessed using the subscript ([]) operator
+// arrays are zero-indexed, meaning they start counting their elements at zero
+// yields "value zero"
+arr[0]:
+// yields "value one"
+arr[1];
+// yields 2
+arr[2];
+// you can change the contents at an index by reassigning with =
+// after this next line, arr will look like ["value zero", "value one", 2, 3]
+arr[3] = 3;
+// you can even store other arrays inside of an array!
+let nested = [["array 0 element 0"], ["array 1 element 0"], ["array 2 element 0"]];
+// you access these values using chained subscripts. the first subscript yields
+// the element stored at that index (which is another array) and the second one
+// subscripts into that yielded array. this yields "array 1 element 0".
+nested[1][0];
+// you can get an array's length using the `.length` property (more on
+// properties in the classes section). this yields 4.
+arr.length;
```

Comment:
Are there any other show-stoppingly important array methods that should be included here?

Reply:
Will let Nivedha or Avery chime in on this.